### ISSUE TITLE
Missing textures in Tinker Station and Tinker's Anvil

### ISSUE THREAD

**Issue by CodePidgy** [SlimeKnights/TinkersConstruct#4744]
### Minecraft Version

1.18.1

### Forge Version

39.0.66

### Mantle Version

1.8.31

### Tinkers' Construct Version

3.4.0.34

### Describe your issue

When using the Tinker Station or the Tinker's Anvil there are a bunch of missing textures. Initially I was using this on my own little modpack and though there was a conflict but it is the exact same with only TC and Mantle installed. 
![image](https://user-images.githubusercontent.com/33547278/153049459-a5124b81-0b2a-4e7f-8df2-6e4d4d0316b2.png)

### Reproduced with only Tinkers' Construct?

Yes

### Confirm the following

- [X] Check if you are not using performance enhancers, such as optifine
- [X] Check if you have searched the tracker to see if your issue already exists, notably pinned issues
- [X] Check if you have look at FAQ to see if your issue is answered
- [ ] Check if you did not read all of the above checkboxes and just checked all the boxes

**Comment by ShinyPorygon** [comment]
Try reinstalling the mod and mantle, not really sure if we can repeat it because under similar condition we've yet to have this happen.

**Comment by CodePidgy** [comment]
That's the thing, in order to check if it was perhaps a conflict, I ended up creating a new profile in CurseForge, so they are reinstalled. Before posting here, I did also do some searching and only managed to find [this](https://www.reddit.com/r/feedthebeast/comments/smee1h/tinkers_construct_118_is_giving_me_missing/) where another person had the same issue. I contacted them to see if they have fixed it but they also haven't found a solution.

**Comment by KnightMiner** [comment]
Provide your full game log please. Preferably on the instance with just tinkers and mantle.

**Comment by castilio20** [comment]
in that link it says that he is using optifine and that is not supported do you use optifine to?

**Comment by CodePidgy** [comment]
Nope, plain old Minecraft. Nothing added except Mantle and TC.

**Comment by KnightMiner** [comment]
> Provide your full game log please. Preferably on the instance with just tinkers and mantle.

**Comment by CodePidgy** [comment]
```
19:17:00.650
Preparing to launch minecraft client for forge-39.0.66
19:17:00.686
Prepared asset index
19:17:08.192
Have 3117 total files to check or download
19:17:08.198
Waiting for 3117 futures
19:17:08.198
Local file D:\Data\Minecraft\Install\versions\forge-39.0.66\forge-39.0.66.jar is incomplete
19:17:53.603
Minecraft client forge-39.0.66 requires 1574664 bytes to download.
19:17:53.604
Starting!
19:17:53.604
Downloading 1 files
19:17:55.758
Download from https://launcher.mojang.com/v1/objects/7e46fb47609401970e2818989fa584fd467cd036/client.jar had invalid checksum! Expected: 7e46fb47609401970e2818989fa584fd467cd036 got: 75ac36b231510d900402ea7cf92afe17aa59cd56
19:18:25.719
Download from https://launcher.mojang.com/v1/objects/7e46fb47609401970e2818989fa584fd467cd036/client.jar passed checksum check.
19:18:25.719
Downloaded https://launcher.mojang.com/v1/objects/7e46fb47609401970e2818989fa584fd467cd036/client.jar to D:\Data\Minecraft\Install\versions\forge-39.0.66\forge-39.0.66.jar
19:18:25.719
Downloads successful
19:18:25.722
TIMER: Downloads took 32 seconds.
19:18:25.722
Downloading Java Runtime java-runtime-beta:17.0.1.12.1
19:18:25.879
Extracting 8 files!
19:18:27.383
Using default game log configuration client-1.12.xml (outputs XML)
19:18:31.886
ModLauncher running: args [--username, DRACL10N, --version, forge-39.0.66, --gameDir, D:\Data\Minecraft\Instances\Tinkers Construct, --assetsDir, D:\Data\Minecraft\Install\assets, --assetIndex, 1.18, --uuid, 37e7956916d24ef8b0abf737766733bd, --accessToken, ❄❄❄❄❄❄❄❄, --clientId, MjUzYmMyMzYtYzFlMy00YTcxLThlZDItZGRiZjU1NmY2N2Vi, --xuid, 2535412362862910, --userType, msa, --versionType, release, --width, 1024, --height, 768, --launchTarget, forgeclient, --fml.forgeVersion, 39.0.66, --fml.mcVersion, 1.18.1, --fml.forgeGroup, net.minecraftforge, --fml.mcpVersion, 20211210.034407]
19:18:53.272
Assets URL 'union:/D:/Data/Minecraft/Install/libraries/net/minecraft/client/1.18.1-20211210.034407/client-1.18.1-20211210.034407-srg.jar%2354!/assets/.mcassetsroot' uses unexpected schema
19:18:53.273
Assets URL 'union:/D:/Data/Minecraft/Install/libraries/net/minecraft/client/1.18.1-20211210.034407/client-1.18.1-20211210.034407-srg.jar%2354!/data/.mcassetsroot' uses unexpected schema
19:18:53.340
Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
19:18:55.035
Setting user: DRACL10N
19:18:55.311
Backend library: LWJGL version 3.2.2 SNAPSHOT
19:18:58.287
Forge mod loading, version 39.0.66, for MC 1.18.1 with MCP 20211210.034407
19:18:58.517
MinecraftForge v39.0.66 Initialized
19:19:08.787
Potentially Dangerous alternative prefix `minecraft` for name `milk`, expected `forge`. This could be a intended override, but in most cases indicates a broken mod.
19:19:08.787
Potentially Dangerous alternative prefix `minecraft` for name `flowing_milk`, expected `forge`. This could be a intended override, but in most cases indicates a broken mod.
19:19:11.135
Narrator library for x64 successfully loaded
19:19:39.441
66 material render infos loaded
19:19:42.034
[forge] Starting version check at https://files.minecraftforge.net/net/minecraftforge/forge/promotions_slim.json
19:19:42.807
Unused frames in sprite tconstruct:block/fluid/obsidian/still: [20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31]
19:19:42.810
Unused frames in sprite tconstruct:block/fluid/stone/still: [20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31]
19:19:43.985
[forge] Found status: BETA_OUTDATED Current: 39.0.66 Target: 39.0.75
19:19:49.444
OpenAL initialized on device OpenAL Soft on Speakers (EasyCall Speakerphone         )
19:19:49.446
Sound engine started
19:19:49.858
Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
19:19:49.909
Created: 256x128x4 minecraft:textures/atlas/signs.png-atlas
19:19:49.913
Created: 512x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
19:19:49.915
Created: 512x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
19:19:49.917
Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
19:19:49.919
Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
19:19:49.919
Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
19:19:54.539
Created: 256x256x0 minecraft:textures/atlas/particles.png-atlas
19:19:54.543
Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
19:19:54.544
Created: 128x128x0 minecraft:textures/atlas/mob_effects.png-atlas
19:20:05.638
Injecting existing registry data into this CLIENT instance
19:20:10.118
Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
19:20:10.120
Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
19:20:10.127
Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
19:20:10.129
Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
19:20:10.131
Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
19:20:10.230
Reloading ResourceManager: Default, TConstruct-1.18.1-3.4.0.34.jar, forge-1.18.1-39.0.66-universal.jar, Mantle-1.18.1-1.8.31.jar
19:20:15.767
Loaded 20 recipes
19:20:19.025
Loaded 1495 advancements
19:20:19.046
Loaded 42 materials in 12.3286 ms
19:20:19.173
126 stats loaded for 61 materials in 125.9734 ms
19:20:19.274
54 traits loaded for 61 materials in 99.8075 ms
19:20:24.528
Preparing start region for dimension minecraft:overworld
19:20:25.969
Preparing spawn area: 0%
19:20:26.159
Preparing spawn area: 0%
19:20:26.315
Preparing spawn area: 0%
19:20:26.316
Preparing spawn area: 0%
19:20:27.157
Preparing spawn area: 0%
19:20:27.330
Preparing spawn area: 0%
19:20:28.163
Preparing spawn area: 0%
19:20:28.269
Preparing spawn area: 0%
19:20:28.802
Preparing spawn area: 0%
19:20:36.153
Preparing spawn area: 0%
19:20:36.153
Preparing spawn area: 0%
19:20:36.154
Preparing spawn area: 0%
19:20:36.155
Preparing spawn area: 0%
19:20:36.157
Preparing spawn area: 0%
19:20:36.350
Preparing spawn area: 0%
19:20:36.897
Preparing spawn area: 3%
19:20:38.041
Preparing spawn area: 6%
19:20:38.041
Preparing spawn area: 6%
19:20:40.060
Preparing spawn area: 9%
19:20:40.060
Preparing spawn area: 9%
19:20:40.064
Preparing spawn area: 9%
19:20:40.068
Preparing spawn area: 9%
19:20:40.366
Preparing spawn area: 11%
19:20:40.745
Preparing spawn area: 15%
19:20:41.332
Preparing spawn area: 23%
19:20:42.295
Preparing spawn area: 31%
19:20:42.295
Preparing spawn area: 31%
19:20:42.761
Preparing spawn area: 35%
19:20:43.259
Preparing spawn area: 54%
19:20:43.719
Preparing spawn area: 62%
19:20:44.483
Preparing spawn area: 68%
19:20:45.069
Preparing spawn area: 74%
19:20:45.816
Preparing spawn area: 80%
19:20:45.817
Preparing spawn area: 80%
19:20:46.926
Preparing spawn area: 83%
19:20:46.927
Preparing spawn area: 83%
19:20:47.220
Preparing spawn area: 83%
19:20:47.732
Preparing spawn area: 83%
19:20:48.349
Preparing spawn area: 89%
19:20:49.370
Preparing spawn area: 89%
19:20:49.408
Preparing spawn area: 89%
19:20:49.936
Preparing spawn area: 91%
19:20:50.296
Preparing spawn area: 92%
19:20:50.737
Preparing spawn area: 94%
19:20:51.267
Preparing spawn area: 98%
19:20:52.615
Time elapsed: 27897 ms
19:20:52.643
Successfully initialized permission handler forge:default_handler
19:20:53.827
Changing view distance to 12, from 10
19:20:53.831
Changing simulation distance to 12, from 0
19:20:54.881
Can't keep up! Is the server overloaded? Running 2087ms or 41 ticks behind
19:20:56.864
Connected to a modded server.
19:21:00.050
Using new advancement loading for net.minecraft.server.PlayerAdvancements@2b9bbef6
19:21:00.075
DRACL10N[local:E:bdeb99d3] logged in with entity id 248 at (-30.572394759349294, 69.0, 81.6027980635459)
19:21:00.243
DRACL10N joined the game
19:21:03.166
Loaded 51 advancements
19:21:09.852
Saving and pausing game...
19:21:10.164
Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
19:21:14.184
Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
19:21:14.186
Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
```

**Comment by KnightMiner** [comment]
I need the client log (as this is a client issue with textures), that looks like a server log. And please use a site such as pastebin to include the log.

**Comment by KnightMiner** [comment]
Is that `latest.log` or `debug.log`? `debug.log` might be more useful here, as I see no information there related to the missing textures. Also, please use a site such as pastebin for the logs to avoid overly long comments

**Comment by CodePidgy** [comment]
How do I manage to find the debug.log? All I can find is the latest.log

**Comment by KnightMiner** [comment]
Should be in the same folder. Are there any files named `.log` besides latest?

Also, are you using regular graphics or fabulous? Doubt that would be the cause, but another issue has me wanting to check there. And do you get missing textures in the part builder? Or in JEI when looking up a part builder recipe?

I am basically trying to find why your computer is different, as I have been using this just fine in a roughly 50 mod modpack on a server without missing textures, and it also works fine in dev.

**Comment by CodePidgy** [comment]
There are no other `.log` files located in the folder with `latest.log`.

I'm using "Fancy" graphics as "Fabulous" is a bit too much for my PC.

All the textures in the Part Builder for selecting which part to make are also missing. The actual parts themselves are fine though, and when put into the Tinker Station, they make replace the missing texture when they are inserted until they are removed again. I believe that the textures that are missing aren't anything you can craft or make, but rather the Stations "representation" of them, if you know what I mean. The icons to show you which parts to use just aren't there.

**Comment by KnightMiner** [comment]
Yes, I know what icons are missing. I was telling you other places they are used as I am trying to find what is not working. The textures work perfectly fine in my testing, so short of you having a performance mod breaking forge that you are not telling me about, I have 0 idea why they are not working.

Please check JEI under the part builder as I told you to see if those are missing, will let me determine if its how the icons are drawn or the icons not getting added in the first place. Also include a screenshot of queens slime tools, as those use a similar system to make their textures work (don't just tell me queens slime looks fine, specifically include a screenshot)

For something else to consider, do you have any resource packs installed?

**Comment by CodePidgy** [comment]
![image](https://user-images.githubusercontent.com/33547278/153298765-b08d3927-ee43-424c-9686-b0785e5adf29.png)
![image](https://user-images.githubusercontent.com/33547278/153299140-6ca7c90d-e473-4879-81e1-837a65d666a0.png)

No resource packs included whatsoever and no performance mods.

**Comment by KnightMiner** [comment]
Queens slime confirms my suspicion, for some reason the resource scanner is not working in your game. Are you playing on a Mac by any chance? Or linux? The only bugs I have seen in the past with this are caused by Mac adding weird files to directories.

**Comment by CodePidgy** [comment]
Nope, WIndows 10. I would like to point out that I've never had this issue before, but to be fair, I've also only really played modpacks that use the older versions of MC and TC.

**Comment by KnightMiner** [comment]
By that do you mean it worked correctly on 1.16?

**Comment by CodePidgy** [comment]
No, only 1.12, unfortunately I've never used that version.

**Comment by KnightMiner** [comment]
Please test in 1.16 when you get a chance. Would help narrow down this bug.

**Comment by CodePidgy** [comment]
I'll do it now and update you

**Comment by CodePidgy** [comment]
1.16.5 works perfectly fine.

**Comment by KnightMiner** [comment]
What launcher are you using?

**Comment by CodePidgy** [comment]
I'm using CurseForge

**Comment by CodePidgy** [comment]
Okay, so, I figured out a solution. Moving the "Default" resource pack above the mod one fixed the issue.

**Comment by KnightMiner** [comment]
Should be fixed in 3.4.1.45.

> Okay, so, I figured out a solution. Moving the "Default" resource pack above the mod one fixed the issue.

This was a race condition, which means doing random things "fixed" it simply by making some section of code faster or slower. That may have worked for you, but it may not work for someone else.
